From fff7db7b11234aa4a71ac56e845df2da0fb84d1b Mon Sep 17 00:00:00 2001
From: Lars Wendler <polynomial-c@gentoo.org>
Date: Fri, 27 Dec 2019 09:54:44 +0100
Subject: [PATCH] Revert "eapply: Drop QA warning for fuzz factor."

This reverts commit 42e7a93971ec55b87d5f290a67ef83fe7454c844.

Revert "eapply: Output verbosely only if patch fails to apply with -F0"

This reverts commit 4a315c41b43867320b27e882bb1427ac553e683c.

Revert "eapply: Drop -s option for patch."

This reverts commit 12d0c48addaf4506dfd7b33f85f2441618a46dd2.

There's no need to accept every stupid brainfart...
---
 bin/phase-helpers.sh | 15 +++------------
 1 file changed, 3 insertions(+), 12 deletions(-)

diff --git a/bin/phase-helpers.sh b/bin/phase-helpers.sh
index 020862ba03..b53d396502 100644
--- a/bin/phase-helpers.sh
+++ b/bin/phase-helpers.sh
@@ -993,20 +993,11 @@ if ___eapi_has_eapply; then
 			ebegin "${prefix:-Applying }${f##*/}"
 			# -p1 as a sane default
 			# -f to avoid interactivity
+			# -s to silence progress output
 			# -g0 to guarantee no VCS interaction
 			# --no-backup-if-mismatch not to pollute the sources
-			local all_opts=(
-				-p1 -f -g0 --no-backup-if-mismatch
-				"${patch_options[@]}"
-			)
-			# try applying with -F0 first, output a verbose warning
-			# if fuzz factor is necessary
-			if ${patch_cmd} "${all_opts[@]}" --dry-run -s -F0 \
-					< "${f}" &>/dev/null; then
-				all_opts+=( -s -F0 )
-			fi
-
-			${patch_cmd} "${all_opts[@]}" < "${f}"
+			${patch_cmd} -p1 -f -s -g0 --no-backup-if-mismatch \
+				"${patch_options[@]}" < "${f}"
 			failed=${?}
 			if ! eend "${failed}"; then
 				__helpers_die "patch -p1 ${patch_options[*]} failed with ${f}"
-- 
2.25.1

